### PR TITLE
feat: add logs message lost - WPB-9221

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9221" title="WPB-9221" target="_blank">BugWPB-9221</a>  Missing messages during sync
  </td></table>
  <br />
 

### Issue

In order to investigate the message lost during sync or the message error in decryption, this PR adds some logs for datadog.

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-data-model/Source/Model/Message/ZMOTRMessage+UpdateEvent.swift [hunk 1]
```diff
@@ -52,7 +52,8 @@ extension ZMOTRMessage {
                 return nil
         }
         zmLog.debug("Processing:\n\(message)")
-
+        let logAttributes: LogAttributes = [LogAttributesKey.eventId.rawValue: updateEvent.safeUUID, LogAttributesKey.nonce.rawValue: updateEvent.messageNonce, LogAttributesKey.messageType.rawValue: updateEvent.safeType]
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/Model/Message/ZMOTRMessage+UpdateEvent.swift]: line break formatting would help me to read:
```suggestion
        let logAttributes: LogAttributes = [
            LogAttributesKey.eventId.rawValue: updateEvent.safeUUID,
            LogAttributesKey.nonce.rawValue: updateEvent.messageNonce,
            LogAttributesKey.messageType.rawValue: updateEvent.safeType
        ]
```

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift [hunk 2]
```diff
@@ -165,8 +166,11 @@ extension EventDecoder {
         sender: UserClient,
         in context: NSManagedObjectContext
     ) {
-        zmLog.safePublic("Failed to decrypt message with error: \(error), client id <\(sender.safeRemoteIdentifier))>")
-        zmLog.error("event debug: \(event.debugInformation)")
+        var attributes: LogAttributes = .safePublic
+        attributes.merge([LogAttributesKey.eventId.rawValue: event.safeUUID,
+                          LogAttributesKey.senderUserId.rawValue: sender.safeRemoteIdentifier.value], uniquingKeysWith: { _, new in new })
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift]:
```suggestion
        attributes.merge(
            [
                LogAttributesKey.eventId.rawValue: event.safeUUID,
                LogAttributesKey.senderUserId.rawValue: sender.safeRemoteIdentifier.value
            ],
            uniquingKeysWith: { _, new in new }
        )
```

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift [hunk 3]
```diff
@@ -382,7 +383,11 @@ extension EventDecoder {
         return events.filter { event in
             // The only message we process arriving in the self conversation from other users is availability updates
             if event.conversationUUID == selfConversationID, event.senderUUID != selfUserID, let genericMessage = GenericMessage(from: event) {
-                return genericMessage.hasAvailability
+                let included = genericMessage.hasAvailability
+                if !included {
+                    WireLogger.updateEvent.warn("dropping stored event", attributes: [LogAttributesKey.eventId.rawValue: event.safeUUID])
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift]: Maybe easier to read?
```suggestion
                    WireLogger.updateEvent.warn(
                        "dropping stored event",
                        attributes: [LogAttributesKey.eventId.rawValue: event.safeUUID]
                    )
```

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift [hunk 4]
```diff
@@ -267,7 +267,9 @@ open class ZMUpdateEvent: NSObject {
     class func eventsArray(with uuid: UUID, payloadArray: [Any]?, transient: Bool, source: ZMUpdateEventSource, pushStartingAt sourceThreshold: UUID?) -> [ZMUpdateEvent] {
 
         guard let payloads = payloadArray as? [[AnyHashable: AnyHashable]] else {
-            zmLog.error("Push event payload is invalid")
+            var attributes: LogAttributes = .safePublic
+            attributes.merge([LogAttributesKey.eventId.rawValue: uuid.transportString().readableHash], uniquingKeysWith: { _, new in new })
```
Comment (markusfassbender) [inline wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: My rule of thumb would be a single value in an array can be one line with `[]` like here, but multiple arguments should have breaks. I think there is even a SwiftLint rule for it, that we don't have activated yet..
```suggestion
            attributes.merge(
                [LogAttributesKey.eventId.rawValue: uuid.transportString().readableHash],
                uniquingKeysWith: { _, new in new }
            )
```

**Review by markusfassbender — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Pull request merged**